src/status_im/contexts/wallet/events.cljs

```diff
@@ -386,3 +389,41 @@
             {:title   title
              :subject title
              :message content})]]}))
+
+(rf/reg-event-fx
+ :wallet/blockchain-status-changed
+ (fn [{:keys [db]} [{:keys [message]}]]
+   (let [chains                  (-> (transforms/json->clj message)
+                                     (update-keys (comp utils.number/parse-int name)))
+         down-chains             (-> (select-keys chains
+                                                  (for [[k v] chains :when (= v "down")] k))
+                                     keys)
+         test-networks-enabled?  (get-in db [:profile/profile :test-networks-enabled?])
+         filter-fn               (fn [chain-id]
+                                   (let [mainnet-chain? (contains? constants/mainnet-chain-ids chain-id)]
+                                     (if test-networks-enabled?
+                                       mainnet-chain?
+                                       (not mainnet-chain?))))
+         chains-filtered-by-mode (remove filter-fn down-chains)
+         chains-down?            (seq chains-filtered-by-mode)
+         chain-names             (when chains-down?
+                                   (->> (map #(-> (utils/id->network %)
+                                                  name
+                                                  string/capitalize)
+                                             chains-filtered-by-mode)
+                                        set
+                                        (string/join ", ")))]
+     (when (seq down-chains)
+       (log/info "[wallet] Chain(s) down: " down-chains)
+       (log/info "[wallet] Test network enabled: " (boolean test-networks-enabled?)))
+
+     (cond-> {:db (assoc-in db [:wallet :statuses :blockchains] chains)}
+
+       chains-down?
+       (assoc :fx
+              [[:dispatch
+                [:toasts/upsert
+                 {:id       :chains-down
+                  :type     :negative
+                  :text     (i18n/label :t/provider-is-down {:chains chain-names})
+                  :duration 10000}]]])))))
```

Comment:
We can extract `:fx` to make the code easier to understand. `:fx` is able to handle `nil`:

```clojure
{:db (assoc-in db [:wallet :statuses :blockchains] chains)
 :fx [(when chains-down?
        [:dispatch
         [:toasts/upsert
          {:id       :chains-down
           :type     :negative
           :text     (i18n/label :t/provider-is-down {:chains chain-names})
           :duration 10000}]])]}
```

Also probably we don't need to keep the vector outside `when`, so you could try:

```clojure
{:db ...
 :fx (when chains-down?
       [[:dispatch ...]])}
```

Reply:
Yes @ulisesmac, It's just I thought why pass nil when we can make use of `cond->`.  We don't have a lot of conditions here.

Reply:
The "new" re-frame API (a vector of Fx able to handle `nil`, was thought to give more flexibility.

I think adding or removing `nil` doesn't change anything.


I was suggesting this because IMO it's easier to "parse"  in developer's mind, since we'd be using simple patterns, I'm not saying `cond->` is bad itself, but I just think it's not needed.

Anyway, this might be a matter of prefernces, so it's ok if you apply it or not :+1:

Reply:
I agree it's easy to read the code. I'm not saying the patterns we used to are wrong. Apologies, if you felt like that 🙏. 

It's just my experiences with some events in the early stages which made me use `cond->` to resolve a bug (encountered due to `nil`). But that was before we upgraded the re-frame and `:fx` is life saver as it can handle `nil` and doesn't throw any error. I will update it 👍